src/hotkeys.c

```diff
@@ -742,8 +742,10 @@ void idle_stdin_cb(EV_P_ ev_idle *w, int revents) {
     }
 
     // Free watchers data
-    free(((stdin_idle_data *)w->data)->data);
-    free(w->data);
+    if (w->data) {
+        free(((stdin_idle_data *)w->data)->data);
+        free(w->data);
```

Comment:
`w->data = NULL;`

Reply:
And whole function uses `((stdin_idle_data *)w->data)`, so maybe it will be useful to assign it to local variable, like:
`stdin_idle_data * idata = ((stdin_idle_data *)w->data);`

Even more - if `ev_idle_stop(EV_A_ w);` can be moved to the top of function, then you can stop the timer and if idata == NULL just return, because all other code depends on it